scripts/install_api.py

```diff
@@ -1,133 +1,204 @@
 #!/usr/bin/env python3
 #
-#    Thanks to @withzombies for letting us adapt his script
+#   Thanks to @withzombies for letting us adapt his script
 #
 
 import importlib.util
 import os
 import sys
 from site import check_enableusersite
+from typing import Optional
 
 if sys.version_info[0] < 3:
-	print("Python 3 required")
-	sys.exit(1)
+    sys.stderr.print("Python 3 is required.")
+    sys.exit(1)
 
 # Handle both normal environments and virtualenvs
 try:
-	from site import getusersitepackages, getsitepackages
+    from site import getusersitepackages, getsitepackages
 except ImportError:
-	from sysconfig import get_path
-	getsitepackages = lambda: get_path('purelib')
-	getusersitepackages = getsitepackages
+    from sysconfig import get_path
+    getsitepackages = lambda: get_path("purelib")
+    getusersitepackages = getsitepackages
+
+
+def print_error(*args, file=None, **kwargs):
+    print(*args, file=sys.stderr, **kwargs)
+
+
+def check_virtual_environment() -> bool:
+    if sys.prefix != sys.base_prefix:
+        if os.environ.get("VIRTUAL_ENV"):
+            return True
+    return False
+
+
+def binaryninja_installed() -> bool:
+    try:
+        binaryninja = importlib.util.find_spec("binaryninja")
+        assert binaryninja is not None
+        binaryninjaui = importlib.util.find_spec("binaryninjaui")
+        assert binaryninjaui is not None
+        return True
+    except ModuleNotFoundError:
+        return False
+    except AssertionError:
+        return False
+
+
+def get_binaryninja_installed_directory() -> Optional[str]:
```

Comment:
Should probably move the error reporting into this function when it still knows what paths it's checking, otherwise you'll get an error like:

```Failed to find installed python expected at None```

Otherwise it looks good and we appreciate the cleanup!

Reply:
I just removed if statement and this function just returns expected path. Also, path will be confirmed at call site.